OpenMTRDemo/MainForm.cs

```diff
@@ -1,44 +1,135 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.ComponentModel;
-using System.Data;
-using System.Drawing;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using System.Windows.Forms;
-using OpenMTR;
+using OpenCvSharp;
 
 namespace OpenMTRDemo
 {
     public partial class MainForm : Form
     {
-        OpenFileDialog FileBrowser;
-        Meter meter;
+        OpenFileDialog OpenBrowser;
+        SaveFileDialog SaveBrowser;
+        OpenMTR.Meter meter;
+
         public MainForm()
         {
-            FileBrowser = new OpenFileDialog();
-            FileBrowser.Filter = "Image Files|*.bmp;*.jpeg;*.jpg;*.gif";
+            OpenBrowser = new OpenFileDialog();
+            SaveBrowser = new SaveFileDialog();
+            OpenBrowser.Filter = "Image Files|*.bmp;*.gif;*.jpg;*.jpeg;*.png";
+            SaveBrowser.Filter = "JPG|*.jpg";
             InitializeComponent();
         }
 
-        private void loadToolStripMenuItem_Click(object sender, EventArgs e)
+        private void openToolStripMenuItem_Click(object sender, EventArgs e)
         {
-            if (FileBrowser.ShowDialog() == DialogResult.OK)
+            if (OpenBrowser.ShowDialog() == DialogResult.OK)
             {
-                meter = ReadData.GetMeter(FileBrowser.FileName);
-                Render();
+                MetaDataTextBox.Text = meter.MeterRead + "";
```

Comment:
You forgot to remove this I am assuming? Throws a Null Reference Exception when loading a file in.